packages/js/product-editor/src/blocks/generic/text-area/edit.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useWooBlockProps } from '@woocommerce/block-templates';
+import { createElement } from '@wordpress/element';
+import { BaseControl } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+import { useInstanceId } from '@wordpress/compose';
+import { BlockControls, RichText } from '@wordpress/block-editor';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { RTLToolbarButton } from './toolbar/toolbar-button-rtl';
+import type {
+	TextAreaBlockEditAttributes,
+	TextAreaBlockEditProps,
+} from './types';
+import AligmentToolbarButton from './toolbar/toolbar-button-alignment';
+
+export function TextAreaBlockEdit( {
+	attributes,
+	setAttributes,
+	context,
+}: TextAreaBlockEditProps ) {
+	const { align, allowedFormats, direction, label, helpText, placeholder } =
+		attributes;
+	const blockProps = useWooBlockProps( attributes, {
+		style: { direction },
+	} );
+
+	const contentId = useInstanceId(
+		TextAreaBlockEdit,
+		'wp-block-woocommerce-product-content-field__content'
+	);
+
+	// `property` attribute is required.
+	const { property } = attributes;
+	if ( ! property ) {
+		throw new Error(
+			__( 'Property attribute is required.', 'woocommerce' )
+		);
+	}
+
+	const [ content, setContent ] = useEntityProp< string >(
+		'postType',
+		context.postType || 'product',
+		property
+	);
+
+	function setAlignment( value: TextAreaBlockEditAttributes[ 'align' ] ) {
+		setAttributes( { align: value } );
+	}
+
+	function changeDirection(
+		value: TextAreaBlockEditAttributes[ 'direction' ]
+	) {
+		setAttributes( { direction: value } );
+	}
+
+	const blockControlsProps = { group: 'block' };
+
+	return (
+		<div className={ 'wp-block-woocommerce-product-text-area-field' }>
+			<BlockControls { ...blockControlsProps }>
+				<AligmentToolbarButton
+					align={ align }
+					setAlignment={ setAlignment }
+				/>
```

Comment:
We should have a way to turn off the alignment toolbar, as for a plain text textarea, it isn't necessary.

Reply:
We need to define this functionality a little bit. It isn't clear to me when/how the block should support rich text or plain text areas.

* Is it the same block that changes its behavior depending on a parameter? (block attribute).
* Could this block support both modes? It could be switchable from the block toolbar.
* Or, should we create two different blocks? (`text-area` and `rich-text`, for instance)

Reply:
Could we address this issue in a follow-up, Matt? I want to avoid increasing the PR size.

Reply:
I'm good if this is addressed in a follow-up.  I think we should allow block attributes that allow us to pick between using a plain textarea vs one with rich text.

Reply:
I agree that it should be switchable between rich text and plain text via a block attribute. I don't think, for our purposes, it makes sense to have it be switchable from the block toolbar.

Reply:
https://github.com/woocommerce/woocommerce/issues/44182